diff -uwr linux-4.8-rc4.new/sound/soc/intel/boards/bytcr_rt5640.c linux-4.8-rc4.orig/sound/soc/intel/boards/bytcr_rt5640.c
--- linux-4.8-rc4.new/sound/soc/intel/boards/bytcr_rt5640.c	2016-09-04 01:29:34.619518130 +0000
+++ linux-4.8-rc4.orig/sound/soc/intel/boards/bytcr_rt5640.c	2016-09-04 01:32:58.820106971 +0000
@@ -389,6 +389,24 @@
 						 BYT_RT5640_SSP0_AIF1),
 
 	},
+	{
+		.callback = byt_rt5640_quirk_cb,
+		.matches = {
+			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "YIFANG"),
+			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "NXW9QC132"),
+		},
+		.driver_data = (unsigned long *)(BYT_RT5640_DMIC2_MAP |
+						 BYT_RT5640_DMIC_EN),
+	},
+	{
+		.callback = byt_rt5640_quirk_cb,
+		.matches = {
+			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "YIFANG"),
+			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "NXW101QC232"),
+	},
+		.driver_data = (unsigned long *)(BYT_RT5640_DMIC2_MAP |
+						 BYT_RT5640_DMIC_EN),
+	},
 	{}
 };
 
